### PR TITLE
Clang-format for Packet++

### PR DESCRIPTION
**Pull request by tigercosmos**
- Applied clang-format to all files under "Packet++"
- Manually checked and fixed the files:
  - [x] all headers
  - [x] all cpps

### CODE REVIEW

**Comment by tigercosmos** [comment]
Manually checked and fixed the files:
- [x] all headers
- [ ] all cpps

**Comment by tigercosmos** [comment]
@seladb  The PR is ready for review. I went through all the files and modified the obvious weird formatting. You can check the commit logs to see the changes.

**Comment by seladb** [comment]
> @seladb The PR is ready for review. I went through all the files and modified the obvious weird formatting. You can check the commit logs to see the changes.

Thank you @tigercosmos ! I will start going over it, but I should warn you that it'll take time to go over 130 files 🙂 
I'll add comments as I see them so you can address them while I go over it

File: Packet++/header/ArpLayer.h [hunk 1]
```diff
@@ -46,8 +48,8 @@ namespace pcpp
 	 */
 	enum ArpOpcode
 	{
-		ARP_REQUEST = 0x0001, ///< ARP request
-		ARP_REPLY   = 0x0002  ///< ARP reply (response)
+		ARP_REQUEST = 0x0001,  ///< ARP request
+		ARP_REPLY = 0x0002     ///< ARP reply (response)
```
Comment (seladb) [inline Packet++/header/ArpLayer.h:52]: I think we need to move the comments on top of every enum value, like we usually do in enums

Comment (clementperon) [reply Packet++/header/ArpLayer.h:52]: I think this is the only doxygen coding style no for enum no ? 

https://stackoverflow.com/questions/13752191/documenting-enum-values-with-doxygen

Comment (tigercosmos) [reply Packet++/header/ArpLayer.h:52]: @seladb I think this is not in the scope of this PR. Feel free to open new issues for non-format-related issues.

Comment (seladb) [reply Packet++/header/ArpLayer.h:52]: @clementperon we document most of our enums in a different way, for example: https://github.com/seladb/PcapPlusPlus/blob/d21c05ccf8cd039a8faef77de02e2f7d7e1549b4/Packet%2B%2B/header/DnsLayerEnums.h#L12

Comment (seladb) [reply Packet++/header/ArpLayer.h:52]: @tigercosmos I agree it doesn't have to be in the scope of this PR. I'll let you decide if you want to fix it or not

File: Packet++/header/Layer.h [hunk 2]
```diff
@@ -29,23 +29,26 @@ namespace pcpp
...
-	 * end of the packet. Here is an example packet showing this concept:
+	 * For example: a pointer to a structured header (e.g tcphdr, iphdr, etc.), protocol header size, payload size,
+	 compute fields that can be automatically computed, print protocol data to string, etc.
+	 * Each protocol instance is obviously part of a protocol stack (which construct a packet). This protocol stack is
+	 represented in PcapPlusPlus in a linked list, and each layer is an element in this list. That's why each layer has
+	 properties to the next and previous layer in the protocol stack.
+	 * The Layer class, as a base class, is abstract and the user can't create an instance of it (it has a private
+	 constructor).
+	 * Each layer holds a pointer to the relevant place in the packet. The layer sees all the data from this pointer
+	 forward until the end of the packet. Here is an example packet showing this concept:
```
Comment (seladb) [inline Packet++/header/Layer.h:51]: For some reason some of the lines don't begin with the `*` character...

**Review by seladb — COMMENTED** [review]
Around 35% reviewed 🙂

File: Packet++/header/Layer.h (hunk 2 above)
Comment (tigercosmos) [reply Packet++/header/Layer.h:51]: When the line is too long, clang-format split it. I manually fixed some, but this one was missed.

File: Packet++/header/DnsResourceData.h [hunk 3]
```diff
@@ -15,59 +15,66 @@
...
-
-	#if __cplusplus > 199711L || _MSC_VER >= 1800 //Maybe this can be 1600 for VS2010
-	#define PCPP_SMART_PTR(T) std::unique_ptr<T>
-	#else
-	#define PCPP_SMART_PTR(T) std::auto_ptr<T>
-	#endif
+	// Visual studio has always been stupid about returning something useful for __cplusplus
+	// Only recently was this fixed - and even then it requires a specific hack to the command line during build
+	// Its easier/more consistent to test _MSC_VER in VS
+	// https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=vs-2017
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/DnsResourceData.h:21]: Sure I can remove it.

File: Packet++/header/DnsResourceData.h [hunk 4]
```diff
@@ -79,95 +86,107 @@ namespace pcpp
...
-
-#if __cplusplus <= 199711L && _MSC_VER < 1800 //Maybe this can be 1600 for VS2010
-		DnsResourceDataPtr(const DnsResourceDataPtr& other) : PCPP_SMART_PTR(IDnsResourceData)((DnsResourceDataPtr&)other) {}
+		explicit DnsResourceDataPtr(IDnsResourceData* ptr) : PCPP_SMART_PTR(IDnsResourceData)(ptr)
+		{}
+
+		// Visual studio has always been stupid about returning something useful for __cplusplus
+		// Only recently was this fixed - and even then it requires a specific hack to the command line during build
+		// Its easier/more consistent to test _MSC_VER in VS
+		// https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=vs-2017
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/DnsResourceData.h:112]: done.

File: Packet++/header/HttpLayer.h [hunk 5]
```diff
@@ -545,26 +574,32 @@ namespace pcpp
...
 
 		/**
-		 * The length of the body of many HTTP response messages is determined by a HTTP header field called "Content-Length". This method sets
-		 * The content-length field value. The method supports several cases:
+		 * The length of the body of many HTTP response messages is determined by a HTTP header field called
+		 * "Content-Length". This method sets The content-length field value. The method supports several cases:
 		 * - If the "Content-Length" field exists - the method will only replace the existing value with the new value
-		 * - If the "Content-Length" field doesn't exist - the method will create this field and put the value in it. Here are also 2 cases:
+		 * - If the "Content-Length" field doesn't exist - the method will create this field and put the value in it.
+		 * Here are also 2 cases:
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/HttpLayer.h:587]: done

File: Packet++/header/IPReassembly.h [hunk 6]
```diff
@@ -8,32 +8,36 @@
...
- *   list
+ * - With each fragment arriving the fragment data is copied right after the previous fragment and the reassembled
+ * packet is gradually being built
+ * - When the last fragment arrives the packet is fully reassembled and returned to the user. Since all fragment data is
+ * copied, the packet pointer returned to the user has to be freed by the user when done using it
+ * - The logic supports out-of-order fragments, meaning that a fragment which arrives out-of-order, its data will be
+ * copied to a list of out-of-order fragments where it waits for its turn. This list is observed each time a new
+ * fragment arrives to see if the next fragment(s) wait(s) in this list
  * - If a non-IP packet arrives it's returned as is to the user
  * - If a non-fragment packet arrives it's returned as is to the user
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/IPReassembly.h:31]: fixed.

File: Packet++/header/IPReassembly.h [hunk 7]
```diff
@@ -43,35 +47,35 @@
...
-	 * - IPReassembly#getCurrentPacket() - get the reassembled data that is currently available, even if reassembly process is not yet completed
-	 * - IPReassembly#removePacket() - remove all data that is currently stored for a packet, including the reassembled data that was gathered
-	 *   so far
+	 * - IPReassembly#processPacket() - process a fragment. This is the main method which should be called whenever a
+	 * new fragment arrives. This method processes the fragment, runs the reassembly logic and returns the result packet
+	 * when it's fully reassembled
+	 * - IPReassembly#getCurrentPacket() - get the reassembled data that is currently available, even if reassembly
+	 * process is not yet completed
+	 * - IPReassembly#removePacket() - remove all data that is currently stored for a packet, including the reassembled
+	 * data that was gathered so far
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/IPReassembly.h]: fixed.

File: Packet++/header/IPReassembly.h [hunk 8]
```diff
@@ -300,139 +358,177 @@ namespace pcpp
...
+		 * @param[out] status An indication of the packet reassembly status following the processing of this fragment.
+		 * Possible values are:
 		 * - The input fragment is not a IPv4 or IPv6 packet
 		 * - The input fragment is not a IPv4 or IPv6 fragment packet
 		 * - The input fragment is the first fragment of the packet
 		 * - The input fragment is not the first or last fragment
-		 * - The input fragment came out-of-order, meaning that wasn't the fragment that was currently expected (it's data is copied to
-		 *   the out-of-order fragment list)
+		 * - The input fragment came out-of-order, meaning that wasn't the fragment that was currently expected (it's
+		 * data is copied to the out-of-order fragment list)
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/IPReassembly.h:412]: fixed.

File: Packet++/header/IPReassembly.h [hunk 9]
```diff
@@ -300,139 +358,177 @@ namespace pcpp
...
+		 * the IP-layer and thus failing to work properly. Default value is ::OsiModelLayerUnknown which means don't
+		 * take this parameter into account
 		 * @return
-		 * - If the input fragment isn't an IPv4/IPv6 packet or if it isn't an IPv4/IPv6 fragment, the return value is a pointer to the input fragment
-		 * - If the input fragment is the last one and the reassembled packet is ready - a pointer to the reassembled packet is
-		 *   returned. Notice it's the user's responsibility to free this pointer when done using it
+		 * - If the input fragment isn't an IPv4/IPv6 packet or if it isn't an IPv4/IPv6 fragment, the return value is a
+		 * pointer to the input fragment
+		 * - If the input fragment is the last one and the reassembled packet is ready - a pointer to the reassembled
+		 * packet is returned. Notice it's the user's responsibility to free this pointer when done using it
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/IPReassembly.h:431]: fixed.

File: Packet++/header/IPReassembly.h [hunk 10]
```diff
@@ -300,139 +358,177 @@ namespace pcpp
...
 		 * @return
-		 * - If the input fragment isn't an IPv4/IPv6 packet or if it isn't an IPv4/IPv6 fragment, the return value is a pointer to a Packet object
-		 *   wrapping the input fragment RawPacket object. It's the user responsibility to free this instance
-		 * - If the input fragment is the last one and the reassembled packet is ready - a pointer to the reassembled packet is
-		 *   returned. Notice it's the user's responsibility to free this pointer when done using it
+		 * - If the input fragment isn't an IPv4/IPv6 packet or if it isn't an IPv4/IPv6 fragment, the return value is a
+		 *pointer to a Packet object wrapping the input fragment RawPacket object. It's the user responsibility to free
+		 *this instance
+		 * - If the input fragment is the last one and the reassembled packet is ready - a pointer to the reassembled
+		 *packet is returned. Notice it's the user's responsibility to free this pointer when done using it
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/IPReassembly.h:473]: fixed.

File: Packet++/header/Layer.h (hunk 2 above)
Comment (tigercosmos) [reply Packet++/header/Layer.h:51]: fixed.

File: Packet++/header/DnsResourceData.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (seladb) [reply Packet++/header/DnsResourceData.h:100]: The comment and the `#if` statement are still there...

File: Packet++/header/Packet.h [hunk 11]
```diff
@@ -34,93 +36,114 @@ namespace pcpp
...
 		/**
-		 * A constructor for creating a packet out of already allocated RawPacket. Very useful when parsing packets that came from the network.
-		 * When using this constructor a pointer to the RawPacket is saved (data isn't copied) and the RawPacket is parsed, meaning all layers
-		 * are created and linked to each other in the right order. In this overload of the constructor the user can specify whether to free
-		 * the instance of raw packet when the Packet is free or not. . This constructor should be used to parse the packet up to a certain layer in the OSI model
+		 * A constructor for creating a packet out of already allocated RawPacket. Very useful when parsing packets that
+		 * came from the network. When using this constructor a pointer to the RawPacket is saved (data isn't copied)
+		 * and the RawPacket is parsed, meaning all layers are created and linked to each other in the right order. In
+		 * this overload of the constructor the user can specify whether to free the instance of raw packet when the
+		 * Packet is free or not. . This constructor should be used to parse the packet up to a certain layer in the OSI
```
Comment (seladb) [inline Packet++/header/Packet.h:111]: nit: two dots `. .`, can we remove one?

File: Packet++/header/SSLLayer.h [hunk 12]
```diff
@@ -129,30 +130,34 @@
...
+ * Cipher suites are named combinations of authentication, encryption, message authentication code (MAC) and key
+ * exchange algorithms used to negotiate the security settings for a network connection using SSL/TLS.
  * There are many known cipher-suites. PcapPlusPlus support above 300 of them, according to this list:
  * http://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml
- * There is a designated class in PcapPlusPlus called pcpp::SSLCipherSuite which represents the cipher-suites and provides
- * access to their attributes. Then there is a static instance of this class for each one of the supported cipher-suites.
- * This means there are 300+ static instances of pcpp::SSLCipherSuite representing the different cipher suites. The user can
- * access them through static methods in pcpp::SSLCipherSuite or from client-hello and server-hello messages where they appear
+ * There is a designated class in PcapPlusPlus called pcpp::SSLCipherSuite which represents the cipher-suites and
+ * provides
```
Comment (seladb) [inline Packet++/header/SSLLayer.h:138]: not sure why clang-format added a line break after the word `provides` 🤔

File: Packet++/header/SSLLayer.h [hunk 13]
```diff
@@ -281,28 +299,29 @@ namespace pcpp
...
-	         /    version    \      |   handshake       \        rest of  |      |          rest      |      |            rest
-	        /     TLS1_0      \            type          \       message      handshake   of fields...   handshake    of fields...
+	          /     |       \               |          \         \               |           \               | \ /
+	 version    \      |   handshake       \        rest of  |      |          rest      |      |            rest /
+	 TLS1_0      \            type          \       message      handshake   of fields...   handshake    of fields...
 	     type                  \    | SSL_SERVER_HELLO    \      fields...|     type                  |     type
-	 SSL_HANDSHAKE           length                   handshake             SSL_CERTIFICATE             SSL_SERVER_KEY_EXCHANGE
-	     (22)                 xxx   |               version,length        |                           |
+	 SSL_HANDSHAKE           length                   handshake             SSL_CERTIFICATE SSL_SERVER_KEY_EXCHANGE (22)
+	 xxx   |               version,length        |                           |
```
Comment (seladb) [inline Packet++/header/SSLLayer.h:318]: clang-format completely messed it up 😕

**Review by seladb — COMMENTED** [review]
60 out of 130 🙂

File: Packet++/header/DnsResourceData.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/header/DnsResourceData.h:100]: I always looked at the wrong line number..., should be fixed now.

File: Packet++/header/Packet.h (hunk 11 above)
Comment (tigercosmos) [reply Packet++/header/Packet.h:111]: nice catch. fixed.

File: Packet++/header/SSLLayer.h (hunk 12 above)
Comment (tigercosmos) [reply Packet++/header/SSLLayer.h:138]: The line was too long. manually modified.

File: Packet++/header/SSLLayer.h (hunk 13 above)
Comment (tigercosmos) [reply Packet++/header/SSLLayer.h:318]: ops, I noticed the above one but missed the below one. fixed.

File: Packet++/header/NdpLayer.h [hunk 14]
```diff
@@ -17,353 +17,393 @@
...
+			return (size_t)m_Data->recordLen * 8;
+		}
+
+		size_t getDataSize() const
+		{
+			if (m_Data == nullptr)
+				return 0;
+
+			return (size_t)m_Data->recordLen * 8 -
+			       (2 * sizeof(uint8_t));  // length value is stored in units of 8 octets
```
Comment (seladb) [inline Packet++/header/NdpLayer.h:82]: nit: if we move the comment up the line will probably not break up

File: Packet++/header/TcpReassembly.h [hunk 15]
```diff
@@ -25,43 +25,61 @@
...
+ * these moments. Existence of this delay is caused by two reasons:
  * - pcpp#TcpReassembly#reassemblePacket() should detect the packets that arrive after the FIN packet has been received
- * - the user can use the information about connections managed by pcpp#TcpReassembly instance. Following methods are used for this purpose: pcpp#TcpReassembly#getConnectionInformation and pcpp#TcpReassembly#isConnectionOpen.
- * Cleaning of memory can be performed automatically (the default behavior) by pcpp#TcpReassembly#reassemblePacket() or manually by calling pcpp#TcpReassembly#purgeClosedConnections in the user code.
- * Automatic cleaning is performed once per second.
+ * - the user can use the information about connections managed by pcpp#TcpReassembly instance. Following methods are
+ * used for this purpose: pcpp#TcpReassembly#getConnectionInformation and pcpp#TcpReassembly#isConnectionOpen. Cleaning
+ * of memory can be performed automatically (the default behavior) by pcpp#TcpReassembly#reassemblePacket() or manually
+ * by calling pcpp#TcpReassembly#purgeClosedConnections in the user code. Automatic cleaning is performed once per
+ * second.
```
Comment (seladb) [inline Packet++/header/TcpReassembly.h:70]: Let's add 2 spaces in the beginning of each line to align with the bullet point?

File: Packet++/header/TcpReassembly.h [hunk 16]
```diff
@@ -72,408 +90,479 @@
...
 		 */
-		OutOfOrderTcpMessageBuffered,
+		enum ReassemblyStatus
+		{
+			/**
+			 * The processed packet contains valid TCP payload, and its payload is processed by `OnMessageReadyCallback`
+			 * callback function. The packet may be:
+			 * 1. An in-order TCP packet, meaning `packet_sequence == sequence_expected`.
+			 *    Note if there's any buffered out-of-order packet waiting for this packet, their associated callbacks
+			 * are called in this `reassemblePacket` call.
```
Comment (seladb) [inline Packet++/header/TcpReassembly.h:308]: Let's add 3 extra spaces at the beginning of this line?

File: Packet++/header/TcpReassembly.h [hunk 17]
```diff
@@ -72,408 +90,479 @@
...
+			 * 2. An out-of-order TCP packet which satisfy `packet_sequence < sequence_expected && packet_sequence +
+			 * packet_payload_length > sequence_expected`. Note only the new data (the `[sequence_expected,
+			 * packet_sequence + packet_payload_length]` part ) is processed by `OnMessageReadyCallback` callback
+			 * function.
```
Comment (seladb) [inline Packet++/header/TcpReassembly.h:312]: ditto

**Review by seladb — COMMENTED** [review]
50% done! 🙂

File: Packet++/header/NdpLayer.h (hunk 14 above)
Comment (tigercosmos) [reply Packet++/header/NdpLayer.h:82]: yes, fixed.

File: Packet++/header/TcpReassembly.h (hunk 15 above)
Comment (tigercosmos) [reply Packet++/header/TcpReassembly.h]: fixed.

File: Packet++/header/TcpReassembly.h (hunk 16 above)
Comment (tigercosmos) [reply Packet++/header/TcpReassembly.h]: fixed.

File: Packet++/header/TcpReassembly.h (hunk 17 above)
Comment (tigercosmos) [reply Packet++/header/TcpReassembly.h]: fixed.

**Review by seladb — COMMENTED** [review]
90 out of 130 files 🙂

File: Packet++/src/HttpLayer.cpp [hunk 18]
```diff
@@ -13,999 +13,981 @@
...
+	struct HttpResponseStatusCodeHash
 	{
-		m_CustomizedMessage = statusMessage;
+		size_t operator()(const HttpResponseStatusCode& status) const
+		{
+			return static_cast<int>(status);
+		}
+	};
+
+	// clang-format off
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/HttpLayer.cpp:560]: Bad format:
image

[the remark this answers is not in the document]
Comment (seladb) [reply Packet++/src/HttpLayer.cpp:560]: What's bad in this format? 🤔

[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/HttpLayer.cpp:560]: First, I don't want it to break the name into the next line. Second, I don't want it to use spaces for each element line.

[the remark this answers is not in the document]
Comment (seladb) [reply Packet++/src/HttpLayer.cpp:560]: Regarding the first - I don't think it's that bad but I'll let you decide.
Regarding the second - adding spaces aligns with all other map initialization we have in our code base...

[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/HttpLayer.cpp:560]: I see. If you are okay for this, I will enable clang-format for this part

[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/HttpLayer.cpp]: done.

File: Packet++/src/TextBasedProtocol.cpp [hunk 19]
```diff
@@ -9,685 +9,702 @@
...
-		m_FieldList = m_FieldList->getNextField();
-	else
-	{
-		curField = m_FieldList;
-		while (curField->getNextField() != fieldToRemove)
-			curField = curField->getNextField();
+		std::pair<std::multimap<std::string, HeaderField*>::iterator,
+		          std::multimap<std::string, HeaderField*>::iterator>
+		    range;
+		range = m_FieldNameToFieldMap.equal_range(fieldName);
```
Comment (seladb) [inline Packet++/src/TextBasedProtocol.cpp:267]: maybe we can apply the same fix we did in line 349 here?

**Review by seladb — COMMENTED** [review]
I'm done reviewing all files! 🎉 

Please see the last few comments

File: Packet++/src/GreLayer.cpp [hunk 20]
```diff
@@ -1,17 +1,17 @@
...
-#include "VlanLayer.h"
+#include "Logger.h"
 #include "MplsLayer.h"
-#include "PayloadLayer.h"
+#include "PPPoELayer.h"
 #include "PacketUtils.h"
-#include "Logger.h"
-#include "EndianPortable.h"
+#include "PayloadLayer.h"
+#include "VlanLayer.h"
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/GreLayer.cpp:14]: This should be an accident.

File: Packet++/src/LdapLayer.cpp [hunk 21]
```diff
@@ -74,88 +77,94 @@ namespace pcpp {
...
+		{ static_cast<uint8_t>(LdapResultCode::SizeLimitExceeded),         LdapResultCode::SizeLimitExceeded         },
+		{ static_cast<uint8_t>(LdapResultCode::CompareFalse),              LdapResultCode::CompareFalse              },
+		{ static_cast<uint8_t>(LdapResultCode::CompareTrue),               LdapResultCode::CompareTrue               },
+		{ static_cast<uint8_t>(LdapResultCode::AuthMethodNotSupported),    LdapResultCode::AuthMethodNotSupported    },
+		{ static_cast<uint8_t>(LdapResultCode::StrongerAuthRequired),      LdapResultCode::StrongerAuthRequired      },
+		{ static_cast<uint8_t>(LdapResultCode::Referral),                  LdapResultCode::Referral                  },
+		{ static_cast<uint8_t>(LdapResultCode::AdminLimitExceeded),        LdapResultCode::AdminLimitExceeded        },
+		// clang-format off
+		{ static_cast<uint8_t>(LdapResultCode::UnavailableCriticalExtension), LdapResultCode::UnavailableCriticalExtension },
+		// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:138]: Sure, fixed.

File: Packet++/src/LdapLayer.cpp [hunk 22]
```diff
@@ -74,88 +77,94 @@ namespace pcpp {
...
+		{ static_cast<uint8_t>(LdapResultCode::ConstraintViolation),       LdapResultCode::ConstraintViolation       },
+		{ static_cast<uint8_t>(LdapResultCode::AttributeOrValueExists),    LdapResultCode::AttributeOrValueExists    },
+		{ static_cast<uint8_t>(LdapResultCode::InvalidAttributeSyntax),    LdapResultCode::InvalidAttributeSyntax    },
+		{ static_cast<uint8_t>(LdapResultCode::NoSuchObject),              LdapResultCode::NoSuchObject              },
+		{ static_cast<uint8_t>(LdapResultCode::AliasProblem),              LdapResultCode::AliasProblem              },
+		{ static_cast<uint8_t>(LdapResultCode::InvalidDNSyntax),           LdapResultCode::InvalidDNSyntax           },
+		{ static_cast<uint8_t>(LdapResultCode::AliasDereferencingProblem), LdapResultCode::AliasDereferencingProblem },
+		// clang-format off
+		{ static_cast<uint8_t>(LdapResultCode::InappropriateAuthentication),  LdapResultCode::InappropriateAuthentication },
+		// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:153]: Fixed.

File: Packet++/src/LdapLayer.cpp [hunk 23]
```diff
@@ -314,11 +330,15 @@ namespace pcpp {
 		for (auto controlRecord : controlsRecord->getSubRecords())
 		{
 			auto controlSequence = controlRecord->castAs<Asn1SequenceRecord>();
+			// clang-format off
 			auto controlType = controlSequence->getSubRecords().at(controlTypeIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+			// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:335]: Sure. fixed.

File: Packet++/src/LdapLayer.cpp [hunk 24]
```diff
@@ -314,11 +330,15 @@ namespace pcpp {
...
 			std::string controlValue;
 			if (controlSequence->getSubRecords().size() > controlValueIndex)
 			{
+				// clang-format off
 				controlValue = controlSequence->getSubRecords().at(controlValueIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+				// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:341]: Fixed.

File: Packet++/src/LdapLayer.cpp [hunk 25]
```diff
@@ -533,24 +577,26 @@ namespace pcpp {
 		}
 		if (authRecord->getSubRecords().size() > saslCredentialsIndex)
 		{
+			// clang-format off
 			auto credentialsAsString = authRecord->getSubRecords().at(saslCredentialsIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+			// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:582]: Fixed.

File: Packet++/src/LdapLayer.cpp [hunk 26]
```diff
@@ -755,24 +833,33 @@ namespace pcpp {
...
+		    ->getValue();
 	}
 
 	std::vector<LdapAttribute> LdapSearchResultEntryLayer::getAttributes() const
 	{
 		std::vector<LdapAttribute> result;
 
+		// clang-format off
 		auto attributes = getLdapOperationAsn1Record()->getSubRecords().at(attributesIndex)->castAs<Asn1SequenceRecord>();
+		// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:855]: Fixed.

File: Packet++/src/LdapLayer.cpp [hunk 27]
```diff
@@ -755,24 +833,33 @@ namespace pcpp {
...
 		for (auto attributeRecord : attributes->getSubRecords())
 		{
 			auto attrAsSequence = attributeRecord->castAs<Asn1SequenceRecord>();
 
+			// clang-format off
 			auto type = attrAsSequence->getSubRecords().at(attributeTypeIndex)->castAs<Asn1OctetStringRecord>()->getValue();
+			// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp:862]: Fixed.

File: Packet++/src/TcpLayer.cpp [hunk 28]
```diff
@@ -28,497 +28,508 @@ namespace pcpp
...
+
+			if (m_PrevLayer->getProtocol() == IPv4)
+			{
+				const IPv4Address srcIP = static_cast<IPv4Layer*>(m_PrevLayer)->getSrcIPv4Address();
+				const IPv4Address dstIP = static_cast<IPv4Layer*>(m_PrevLayer)->getDstIPv4Address();
+
+				// clang-format off
+				checksumRes = pcpp::computePseudoHdrChecksum(reinterpret_cast<uint8_t*>(tcpHdr), getDataLen(),
+				                                             IPAddress::IPv4AddressType, PACKETPP_IPPROTO_TCP, srcIP, dstIP);
+				// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/TcpLayer.cpp:286]: Well, I am not a fan of it though.

File: Packet++/src/TextBasedProtocol.cpp (hunk 19 above)
Comment (tigercosmos) [reply Packet++/src/TextBasedProtocol.cpp:267]: Fixed.

File: Packet++/src/SdpLayer.cpp [hunk 29]
```diff
@@ -10,134 +9,137 @@
...
+		addField(PCPP_SDP_PROTOCOL_VERSION_FIELD, "0");
+
+		std::stringstream sessionIDStream;
+		sessionIDStream << sessionID;
+		std::stringstream sessionVersionStream;
+		sessionVersionStream << sessionVersion;
+		std::string networkInfo = "IN IP4 " + ipAddress.toString();
+		// clang-format off
+		std::string originatorFieldValue = username + " " + sessionIDStream.str() + " " + sessionVersionStream.str() + " " + networkInfo;
+		// clang-format on
```
[the remark this answers is not in the document]
Comment (tigercosmos) [reply Packet++/src/SdpLayer.cpp]: Fixed

File: Packet++/src/LdapLayer.cpp [hunk 30]
```diff
@@ -524,7 +562,9 @@ namespace pcpp {
 			throw std::invalid_argument("Authentication type is not sasl");
 		}
 
+		// clang-format off
 		auto authRecord = getLdapOperationAsn1Record()->getSubRecords().at(credentialIndex)->castAs<Asn1ConstructedRecord>();
+		// clang-format on
```
Comment (seladb) [inline Packet++/src/LdapLayer.cpp:567]: We don't need to turn off clang-format here

**Review by seladb — APPROVED** [review]
Please see one last comment, otherwise LGTM!

File: Packet++/src/LdapLayer.cpp (hunk 30 above)
Comment (tigercosmos) [reply Packet++/src/LdapLayer.cpp]: Fixed.

**Pull request merged**